Automattic/jetpack
# Search: use `config` package to initialize search feature

Fixes #23150

#### Changes proposed in this Pull Request:
- Centralize search initializing logic to `Intializer`
- Add search initialization to the `config` package
- Use `config` to initialize search in Jetpack

#### Jetpack product discussion
pcNPJE-Ch-p2

#### Does this pull request change what data or activity we track or use?
no


#### Testing instructions:
* Go to a site with Search subscription
* Ensure Instant Search, Classic Search, Search Dashboard, Customizer, JP Search Widgets and Customberg work well
* Ensure all tests pass
